test/es-module/test-esm-loader-hooks.mjs

```diff
@@ -0,0 +1,72 @@
+// Flags: --expose-internals
+import '../common/index.mjs';
+import ESMLoader from '../../lib/internal/modules/esm/loader.js';
+import assert from 'assert';
+
+(function testHookArgs() {
```

Comment:
Why use a function?

Reply:
There will conceivably be more of these, so using the wrapper avoids collisions and facilitates a fresh state for each set (ex if a subsequent set of cases does not want the custom loader added here—which it probably doesn't).

Now that I think about it, I vaguely remember some new fanciness added in a recent ECMAScript version that will create an isolated scope without the function wrapper. Or maybe it was a proposal?

Reply:
You can use a block to create a scope for `let/const` variables:

```js
{
  // Not available outside of the block
  const esmLoader = new ESMLoader();
}
```

Reply:
ah cool. that's what i was thinking of. thanks!